File: src/gestorAplicacion/economia/Ingreso.java
package gestorAplicacion.economia;

import java.time.LocalDate;

import gestorAplicacion.usuario.Cuenta;

public class Ingreso extends Movimiento{

	/**
	 * 
	 */
	private static final long serialVersionUID = -3919824199311137700L;
	
	private Cuenta cuentaOrigen;
	private Cuenta cuentaDestino;
	private Divisa divisaOrigen;
	private Divisa divisaDestino;
	
	public Ingreso(double valorDestino, double valorOrigen, LocalDate fechaCreacion,Cuenta cuentaOrigen, Cuenta cuentaDestino, Divisa divisaOrigen, Divisa divisaDestino) {
		this(valorDestino, valorOrigen, fechaCreacion, true,null, cuentaOrigen, cuentaDestino,divisaOrigen,divisaDestino);
	}

	public Ingreso(double valorDestino, LocalDate fechaCreacion , Banco bancoOrigen, Cuenta cuentaDestino, Divisa divisa) {
		this(valorDestino,valorDestino,fechaCreacion,false, bancoOrigen, null,cuentaDestino,divisa,divisa);
	}

	public Ingreso(double valorDestino, double valorOrigen, LocalDate fechaCreacion, boolean interno, Banco banco, Cuenta cuentaOrigen, Cuenta cuentaDestino, Divisa divisaOrigen, Divisa divisaDestino) {
		super(valorDestino, valorOrigen, fechaCreacion, interno, banco);
		this.cuentaOrigen = cuentaOrigen;
		this.cuentaDestino = cuentaDestino;
		this.divisaOrigen = divisaOrigen;
		this.divisaDestino = divisaDestino;
	}

	public Cuenta getCuentaOrigen() {
		return cuentaOrigen;
	}
	public void setCuentaOrigen(Cuenta cuentaOrigen) {
		this.cuentaOrigen = cuentaOrigen;
	}
	public Cuenta getCuentaDestino() {
		return cuentaDestino;
	}
	public void setCuentaDestino(Cuenta cuentaDestino) {
		this.cuentaDestino = cuentaDestino;
	}
	public Divisa getDivisaOrigen() {
		return divisaOrigen;
	}
	public void setDivisaOrigen(Divisa divisaOrigen) {
		this.divisaOrigen = divisaOrigen;
	}
	public Divisa getDivisaDestino() {
		return divisaDestino;
	}
	public void setDivisaDestino(Divisa divisaDestino) {
		this.divisaDestino = divisaDestino;
	}
		
}

File: Python/gestorAplicacion/economia/Ingreso.py
from gestorAplicacion.usuario import Movimiento

class Ingreso(Movimiento):

    def __init__(self, valorDestino, valorOrigen, fechaCreacion, cuentaOrigen, cuentaDestino, divisaOrigen, divisaDestino, banco):
        super().__init__(valorDestino, valorOrigen, fechaCreacion, True, banco)
        self._cuentaOrigen = cuentaOrigen
        self._cuentaDestino = cuentaDestino
        self._divisaOrigen = divisaOrigen
        self._divisaDestino = divisaDestino

    def getCuentaOrigen(self):
        return self._cuentaOrigen

    def setCuentaOrigen(self, cuentaOrigen):
        self._cuentaOrigen = cuentaOrigen

    def getCuentaDestino(self):
        return self._cuentaDestino

    def setCuentaDestino(self, cuentaDestino):
        self._cuentaDestino = cuentaDestino

    def getDivisaOrigen(self):
        return self._divisaOrigen

    def setDivisaOrigen(self, divisaOrigen):
        self._divisaOrigen = divisaOrigen

    def getDivisaDestino(self):
        return self._divisaDestino
        
    def setDivisaDestino(self, divisaDestino):
        self._divisaDestino = divisaDestino

